src/sentry/models/rulesnooze.py

```diff
@@ -0,0 +1,32 @@
+from django.db import models
+from django.db.models import CheckConstraint, Q
+
+from sentry.db.models import Model, region_silo_only_model, sane_repr
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+@region_silo_only_model
+class RuleSnooze(Model):
+    """
+    Duration an issue alert or metric alert is snoozed for a user. Null `until` value means snoozed forever.
+    """
+
+    __include_in_export__ = True
+
+    user = HybridCloudForeignKey("sentry.User", unique=True, on_delete="CASCADE")
+    rule = HybridCloudForeignKey("sentry.Rule", null=True, unique=True, on_delete="CASCADE")
+    alert_rule = HybridCloudForeignKey("sentry.AlertRule", null=True, on_delete="CASCADE")
```

Comment:
I think we only need to use the hybrid cloud key on `user`. The other relations will be in region silos and so should be fine to use a real foreign key

Reply:
oh lol

Reply:
Yes, only the user will need HybridCloudForeignKey(). That relation class ensures that when a user is deleted and a cross-region message (outbox) is recieved that we can also remove all the data from the region silo that the deleted user had created. This lets us simulate foreign keys but with eventual consistency.